src/Controller/Component/DataTablesComponent.php

```diff
@@ -199,12 +199,12 @@ private function _addCondition($column, $value, $type = 'and')
         $columnName = explode('.', $column)[1]; // Column names always come in the table.field form
         $fieldType = TableRegistry::get($this->_tableName)->schema()->column($columnName)['type']; // standarized ORM type
         $validTypesForCaseConversion = ['string', 'text']; // valid types to do case insensitive comparison (lower())
-
-        //$value = $this->config('search.ignoreCase') ? strtolower($value) : $value;
         $right = $this->config('search.prefix') ? "{$value}%" : "%{$value}%";
 
         if ($this->config('search.ignoreCase') && in_array($fieldType, $validTypesForCaseConversion)) {
-          $condition = ["LOWER({$column}) LIKE" => "LOWER('{$right}')"];
+          $query = TableRegistry::get($this->_tableName)->find(); // Get table reference
```

Comment:
I think here it is better to do
`$query = TableRegistry::get($this->_tableName)->query();`